File: backend/src/schemas/diary.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class AttendeeInfo(BaseModel):
    user_id: int
    name: str
    profile_image_url: Optional[str] = None


class DiaryCreate(BaseModel):
    group_id: int
    user_id: Optional[int] = None  # 유저 정보도 저장
    schedule_id: Optional[int] = None
    attendance_id: Optional[int] = None
    diary_text: str
    image_path: Optional[str] = None


class DiaryRead(BaseModel):
    id: int
    group_id: int
    user_id: int
    schedule_id: Optional[int]
    attendance_id: Optional[int]
    title: str
    diary_text: str
    summary: Optional[str] = None
    image_path: Optional[str]
    created_at: datetime

    hashtags: Optional[str] = None
    attendees: List[AttendeeInfo] = []

    class Config:
        from_attributes = True

File: frontend/src/types/diary.ts
export interface AttendeeInfo {
  user_id: number;
  name: string;
  profile_image_url?: string;
}

export interface Diary {
  id: number;
  group_id: number;
  user_id: number;
  schedule_id?: number;
  attendance_id?: number;
  title: string;
  diary_text: string;
  summary?: string;
  image_path?: string;
  created_at: string;
  hashtags?: string[];
  attendees: AttendeeInfo[];
}

export interface DiaryAutoGenerateInput {
  group_id: number;
  schedule_id: number;
  attendance_id: number;
  user_id: number;
}
